src/types/mod.rs

```diff
@@ -226,8 +227,36 @@ impl days_ms {
     }
 }
 
+const MS_IN_DAY: i32 = 86_400_000;
+
+impl Ord for days_ms {
+    fn cmp(&self, other: &Self) -> Ordering {
+        let self_days = self.days() + self.milliseconds() / MS_IN_DAY;
```

Comment:
an alternative implementation is to wrap the milliseconds on `new` instead of on equality. I.e. `new(1,MS_IN_DAY)` is stored as `[2,0]` instead of `[1,MS_IN_DAY]`. Not sure which is better. Same applies for NS in `MonthsDaysNs`